shapely/strtree.py

```diff
@@ -104,47 +104,46 @@ def __init__(
             ShapelyDeprecationWarning,
             stacklevel=2,
         )
-        self._tree = None
         self.node_capacity = node_capacity
-        self._rev = {
-            item: geom
-            for geom, item in self._iterinitdata(
-                zip(geoms, items) if items is not None else geoms
-            )
-            if not geom.is_empty
-        }
-
-        if self._rev:
-            self._init_tree(self._rev.items())
-
-    def _iterinitdata(
-        self,
-        initdata: Union[Iterable[Tuple[BaseGeometry, Any]], Iterable[BaseGeometry]],
-    ) -> Iterator[Tuple[BaseGeometry, Any]]:
-        if not initdata:
-            return
-
-        for enum_idx, item in enumerate(initdata):
-            if isinstance(item, tuple):
-                yield item[0], item[1]
-            elif isinstance(item, BaseGeometry):
-                yield (item, enum_idx)
-
-    def _init_tree(self, rev_initdata: ItemsView[Any, BaseGeometry]):
-        if rev_initdata:
-            self._tree = lgeos.GEOSSTRtree_create(self.node_capacity)
-            for item, geom in rev_initdata:
-                lgeos.GEOSSTRtree_insert(self._tree, geom._geom, ctypes.py_object(item))
-
-    def __getstate__(self):
-        state = self.__dict__.copy()
-        del state["_tree"]
-        return state
-
-    def __setstate__(self, state):
-        self.__dict__.update(state)
-        if self._rev:
-            self._init_tree(self._rev.items())
+
+        if geoms is None:
+            # TODO should we allow this?
+            geoms = []
+
+        geoms = list(geoms)
+        if geoms and isinstance(geoms[0], tuple):
+            # TODO should we allow this?
+            # unzip the list of tuples
+            geoms, items = list(zip(*geoms))
```

Comment:
Those two TODO items could be removed after https://github.com/Toblerity/Shapely/pull/1172